src/constructs/loadbalancing/alb.test.ts

```diff
@@ -195,27 +168,8 @@ describe("The GuApplicationListener class", () => {
       certificates: [{ certificateArn: "" }],
     });
 
-    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
-    expect(Object.keys(json.Resources)).not.toContain("ApplicationListener");
-  });
-
-  test("overrides the id if the stack migrated value is true", () => {
-    const stack = simpleGuStackForTesting({ migratedFromCloudFormation: true });
-
-    const loadBalancer = new GuApplicationLoadBalancer(stack, "ApplicationLoadBalancer", { vpc });
-    const targetGroup = new GuApplicationTargetGroup(stack, "GrafanaInternalTargetGroup", {
-      vpc: vpc,
-      protocol: ApplicationProtocol.HTTP,
-    });
-
-    new GuApplicationListener(stack, "ApplicationListener", {
-      loadBalancer,
-      defaultAction: ListenerAction.forward([targetGroup]),
-      certificates: [{ certificateArn: "" }],
-    });
-
-    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
-    expect(Object.keys(json.Resources)).toContain("ApplicationListener");
+    expect(stack).not.toHaveResourceOfTypeAndLogicalId("AWS::ElasticLoadBalancingV2::Listener", "AppListener");
+    expect(stack).toHaveResourceOfTypeAndLogicalId("AWS::ElasticLoadBalancingV2::Listener", /ApplicationListener.+/);
   });
 
   test("does not override the id if the stack migrated value is true but the override id value is false", () => {
```

Comment:
I think this is similar to the test(s) above: https://github.com/guardian/cdk/pull/400/files#r609853771

Reply:
Following our discussion above, it seems like it's worth keeping this test. If so, we might want to update the name here too:

`"does not override the id if the stack migrated value is true but the existingLogicalId prop is not defined"`